src/panels/config/entities/entity-registry-settings-editor.ts

```diff
@@ -609,32 +616,48 @@ export class EntityRegistrySettingsEditor extends LitElement {
             @selected=${this._switchAsChanged}
             @closed=${stopPropagation}
           >
-            <ha-list-item
-              value="switch"
-              .selected=${!this._deviceClass || this._deviceClass === "switch"}
-            >
-              ${this.hass.localize(
-                "ui.dialogs.entity_registry.editor.device_classes.switch.switch"
-              )}
-            </ha-list-item>
-            <ha-list-item
-              value="outlet"
-              .selected=${this._deviceClass === "outlet"}
-            >
-              ${this.hass.localize(
-                "ui.dialogs.entity_registry.editor.device_classes.switch.outlet"
-              )}
-            </ha-list-item>
+            ${domain === "switch"
+              ? html`<ha-list-item
+                    value="switch"
+                    .selected=${!this._deviceClass ||
+                    this._deviceClass === "switch"}
+                  >
+                    ${this.hass.localize(
+                      "ui.dialogs.entity_registry.editor.device_classes.switch.switch"
+                    )}
+                  </ha-list-item>
+                  <ha-list-item
+                    value="outlet"
+                    .selected=${this._deviceClass === "outlet"}
+                  >
+                    ${this.hass.localize(
+                      "ui.dialogs.entity_registry.editor.device_classes.switch.outlet"
+                    )}
+                  </ha-list-item>`
+              : html`<ha-list-item
+                    value="switch"
+                    .selected=${this._switchAs === "switch"}
+                  >
+                    ${domainToName(this.hass.localize, "switch")}
+                  </ha-list-item>
+                  <ha-list-item
+                    value="switch"
+                    .selected=${this._switchAs === domain}
+                  >
+                    ${domainToName(this.hass.localize, domain)}
+                  </ha-list-item>`}
             <li divider role="separator"></li>
             ${this._switchAsDomainsSorted(
               SWITCH_AS_DOMAINS,
               this.hass.localize
-            ).map(
-              (entry) => html`
-                <ha-list-item .value=${entry.domain}>
-                  ${entry.label}
-                </ha-list-item>
-              `
+            ).map((entry) =>
+              this._switchAs === entry.domain
+                ? nothing
+                : html`
+                    <ha-list-item .value=${entry.domain}>
+                      ${entry.label}
+                    </ha-list-item>
+                  `
```

Comment:
Should we display other options when `this._switchAs` is defined because it's not possible to convert a light to a cover, right ?

I got a empty field when I select cover.

https://github.com/home-assistant/frontend/assets/5878303/e63d5836-e278-4bf6-acd7-f3c3a7f13229